pkg/utils/tls.go

```diff
@@ -245,6 +245,9 @@ func NewGrpcServer(params ServerParams) (*grpc.Server, error) {
 		config := &tls.Config{
 			Certificates: []tls.Certificate{serverCert},
 			ClientAuth:   tls.NoClientCert,
+			CipherSuites: []uint16{
+				tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
```

Comment:
We should pick this cipher suite from the config YAML file instead of hard-coding here. That way, we don't enforce any particular cipher suite and user can change it easily, if they ever want to, without having to touch the code.

Reply:
In addition to this cipher suite for TLS 1.2, the crypto reviewer has suggested that we also add TLS_AES256_GCM_SHA384 cipher suite for TLS 1.3. And have TLS 1.3 higher up in priority, if applicable.

Reply:
The TLS version used is decided by the golang's "crypto/tls" package. The description says it partially implements TLS 1.2 and TLS 1.3.

Regarding the ciphersuite, what are the probable cases where the user has to change the ciphers? As the communication is only between agent and the manager and we own the modules, I believe we can keep the implementation details abstracted from the user.